### PR TITLE
Some ci.yml updates

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
Resolves: https://github.com/starbeamjs/starbeam/issues/8

Followup for lint / project config fixes here: https://github.com/NullVoxPopuli/starbeam/pull/1 (wip)

added changeset action, within the publish command defined, it'll open a PR
    
the lint command revealed a good number of errors, so those will need to
be resolved.
    
in the ci workflow,
      generate list of workspaces for if we need to iterate over via the "matrix"
      config. As the project grows, it may be beneficial to switch to this
      as logs will get long, and it'll be harder to tell where errors to
      com from. This approach, used by embroider for "scenarios" allows
      seeing what failed at a glance, without having to dig through logs.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -3,7 +3,41 @@ on:
   push:
     branches: [master]
   pull_request:
+
+concurrency: ${{ github.workflow }}-${{ github.ref }}
+
 jobs:
+  install_dependencies:
```
Comment (NullVoxPopuli) [inline .github/workflows/ci.yml:10]: this mostly primes the cache, which the other tasks than utilize

**Comment by wycats** [comment]
@NullVoxPopuli do you think it's good to merge this (most of the issues already existed before the PR and it might not be worth letting it get stale to resolve them) and then follow up quickly to address the lints?

Or do you want to hold off until the lints are addressed?

**Comment by NullVoxPopuli** [comment]
I think it's good to move forward -- I'm investigating the lints in a separate branch and learning about ts-aware lint rules

**Comment by NullVoxPopuli** [comment]
Just kidding, we should be all green now (I finished the first pass at getting the lints running).

File: .eslintrc.json [hunk 2]
```diff
@@ -1,14 +1,13 @@
 {
   "root": true,
-  "parser": "@typescript-eslint/parser",
```
Comment (NullVoxPopuli) [inline .eslintrc.json:3]: using overrides for various types of files is much easier to maintain, and is the whole premise of https://github.com/NullVoxPopuli/eslint-configs (and how `configs.ember()` can be TS, JS, App, Addon, v2 Addon, all in one)

File: .eslintrc.json [hunk 3]
```diff
@@ -23,56 +22,79 @@
...
-    ],
     "unused-imports/no-unused-imports": "error",
     "no-empty": "warn",
     "no-fallthrough": "off",
     "prefer-const": "off",
     "prefer-arrow-callback": "warn"
   },
   "overrides": [
+    {
+      "files": ["**/*.{ts,tsx}"],
```
Comment (NullVoxPopuli) [inline .eslintrc.json:43]: it turned out a good chunk of the errors in the lint command about tsconfig.json / project were due to running these lints on file extensions that were not ts or tsx as those other files are excluded from the tsconfig.json (js, d.ts, etc)

File: .eslintrc.json (hunk 2 above)
Comment (wycats) [reply .eslintrc.json:3]: Yeah, this is totally fine with me.

File: .eslintrc.json (hunk 3 above)
Comment (wycats) [reply .eslintrc.json:43]: Great catch!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (wycats) [reply .github/workflows/ci.yml:10]: We can keep an eye on this going forward. It seems like, at least for now, either way is fine.

**Review by wycats — APPROVED** [review]
Looks good to me!

**Pull request merged**